### PR TITLE
Advance Releasing Configuration

### PR DESCRIPTION
**Pull request by earthchie**
Sometimes filter desired pokemon with ``-iv`` is not enough. Some Pokemons have decent stats but their potential value is low because how the value distribted between stamina, attack, and defense. E.g. Charmeleon with IVs 15/15/0, its potential value is 0.67 which quite low - the truth is this dude is badass, you don't really need to care much about defense for this guy. He is fragile anyway xD
   
To config the whitelist, copy file ``pkmn_whitelist.json.example`` and rename it to  ``pkmn_whitelist.json``
Edit this file however you like, but keep in mind:

1. Pokemon name is always capitalize and case-sensitive
2. These settings will override ``-c`` and ``-iv``. Especially field ``any``, so use with caution.

image

### CODE REVIEW

**Comment by earthchie** [comment]
My English may not good. feel free to edit README.md :)

**Comment by skydrome** [comment]
i like this, can also be used just for keeping pokemon you want to evolve

**Comment by douglascamata** [comment]
This code has potential to replace all our release logic if you just add `min_iv`in the configuration.

**Comment by douglascamata** [comment]
And add a `smart: 30` field for when you want to keep just one type of that Pokemon, but only release if CP is lower than 30.

**Comment by earthchie** [comment]
> This code has potential to replace all our release logic if you just add min_ivin the configuration.

@douglascamata yes. I intend to make it work this way, that's why I mention it in the readme

> 2. These settings will override -c and -iv. Especially field any, so use with caution.

may be I should remove min_cp options?

**Comment by douglascamata** [comment]
@earthchie rebase your branch from dev right now, just to ensure. I'll not allow anyone to merge any release logic modification. 

1. Remove all other release logic in favour of yours 
2. Add what we will call `smart` release option, that works as I told you (is smart release is 200, keep just one copy of every Pokemon kind, but don't release anything above CP 200)

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py [hunk 1]
```diff
@@ -45,9 +45,9 @@ def work(self):
                                     'individual_attack', 'individual_defense', 'individual_stamina']
                                 for individual_stat in iv_stats:
                                     try:
-                                        total_IV += pokemon[
-                                            'pokemon_data'][individual_stat]
+                                        total_IV += pokemon['pokemon_data'][individual_stat]
                                     except:
+                                        pokemon['pokemon_data'][individual_stat] = 0
                                         continue
                                 print total_IV
```
Comment (sinistance) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: can you pls also comment out this line. because it printed weird number 😄

**Comment by earthchie** [comment]
lol damn. too hard for first-day-how-to-Python like me. 
May be you should open issue and ask people to help.

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (earthchie) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: I don't know who add that. also curious myself lol

but yes. I'll remove it.

**Comment by douglascamata** [comment]
@earthchie give me write permissions to your fork on the project and I'll do it for you.

**Comment by earthchie** [comment]
invited

**Comment by solderzzc** [comment]
conflicts....

**Comment by brantje** [comment]
Might be due the new logging

**Comment by earthchie** [comment]
dont pull yet.

**Comment by earthchie** [comment]
@solderzzc conflict solved. test. should be fine to merge now.

**Comment by douglascamata** [comment]
Not yet, buddies. I'm improving this.

**Comment by douglascamata** [comment]
btw, I already had commits solved locally, @earthchie

**Comment by douglascamata** [comment]
ok, now I have tons of conflicts.

**Comment by earthchie** [comment]
oh I though you will make another PR. alright. thanks.

**Comment by solderzzc** [comment]
Should this commit be merged? @douglascamata @earthchie @skydrome

**Comment by earthchie** [comment]
not yet. wait for improve version from @douglascamata

**Comment by earthchie** [comment]
@douglascamata I see you released new logic now. But your logic work quite different from mine.

I have question: what if I want to keep pokemon with max IV only in attack?

I mean 0/15/0 is fine on some Pokemon like Charlizard because it fragile anyway.
Same as some tank Pokemon like Vaporeon I'd prefer 15/0/15 and this logic prevent me from doing so.

This potential value is kinda useless when you hunt for competition Pokemon.

Would you consider to add this feature in the future please?

**Comment by solderzzc** [comment]
Any update here?

**Comment by douglascamata** [comment]
@earthchie if you want that, just use my base to add configuration in json and in the proper method. Should be easy to add new release logic based on configuration now.

**Comment by solderzzc** [comment]
cleaning up the pull-request folder, please reopen when you done the rebase.

**Pull request closed without merging**